spock-spring/boot2-test/boot2-test.gradle

```diff
@@ -15,8 +17,18 @@
  */
 
 plugins {
-  id "io.spring.dependency-management" version "1.0.5.RELEASE"
-  id 'org.springframework.boot' version "2.5.3"
+  // Normally, the Spring Boot 2 Gradle Plugin would not be compatible with Gradle 9+
+  // To get around this issue, we use the dependency-management plugin in isolation
+  // as described here: https://docs.spring.io/spring-boot/docs/2.7.7/gradle-plugin/reference/htmlsingle/#managing-dependencies.dependency-management-plugin.using-in-isolation
+  // This way, we get proper test execution and realistic version resolution on Gradle 9
+  id 'org.springframework.boot' version "2.7.18" apply false
+  id "io.spring.dependency-management" version "1.1.7"
```

Comment:
How about instead removing the plugin from both places? 
It is a relict from times when Gradle did not have built-in BOM support, by now does more harm than good, and even its maintainer recommends not to use it anymore, but instead the built-in BOM support using `platform(...)`.
@leonard84 what do _you_ think?